### PR TITLE
Validate boolean values as is

### PR DESCRIPTION
**Pull request by benmccallum**
Suggestion for: https://github.com/jeffhandley/strickland/issues/15

### CODE REVIEW

File: strickland/src/minLength.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ export default function minLengthValidator(validatorProps) {
                 minLength: validatorProps
             };
         } else {
-            props = validatorProps
+            props = validatorProps;
```
Comment (jeffhandley) [inline strickland/src/minLength.js]: Wow -- that was quite the copy pasta I had there.  Thanks for cleaning that up.

File: strickland/src/compare.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ export default function compareValidator(validatorProps) {
 
         let isValid = true;
 
-        if (isFalsyButNotZero(value)) {
+        if (typeof compare !== 'boolean' && isFalsyButNotZero(value)) {
```
Comment (jeffhandley) [inline strickland/src/compare.js]: Would you mind adding a couple unit tests for this behavior?  And if it's not too much to ask, updating the docs for `compare` to call this handling out as well.

**Comment by benmccallum** [comment]
Adding tests with my original code change made me find another weird scenario. Validating `false`or `true` value with a `compare`(3)` returned `{ isValid: false }`. 

I ended up reverting my original chane and instead changing the utility method `isFalsyButNotZero` to `isEmptyValue`, matching the comment where it's used (`// Empty values are always valid except with the required validator`) and updating the implementation to treat `boolean` values as non-empty too. 

This method is used in the `min`, `max` and `compare` validators, so I confirmed existing tests still passed and added some more specifically around these cases.

* My issue was with the `compare` validator, which I can see someone running into the same scenario with. 
* I can't see someone using the `min` or `max` validators to test a `boolean` value, but in any case now those tests will `now return a failure, just as they do if a `string` value is passed.

If you do however want the `min` and `max` validators to treat `false` as valid, we may just need to re-introduce the `isFalsyButNotZero` method for them specifically.

File: strickland/test/max.spec.js [hunk 3]
```diff
@@ -41,6 +41,16 @@ describe('max', () => {
...
+
+        it('with a boolean false value, it is not valid', () => {
+            const result = validate(false);
+            expect(result.isValid).toBe(false);
+        });
+
+        it('with a boolean true value, it is not valid', () => {
+            const result = validate(true);
+            expect(result.isValid).toBe(false);
+        });
```
Comment (jeffhandley) [inline strickland/test/max.spec.js]: Thanks for adding these tests!

**Comment by jeffhandley** [comment]
This fix was included in Strickland v2.0.1, which has been published to NPM.

Thanks for the first community bug fix, @benmccallum!! :tada:

**Pull request merged**